utils/file_operations.py

```
@@ -0,0 +1,281 @@
+        return False, []
+
+    # Build set of resolved file paths for O(1) lookup
+    file_paths_set = {path.resolve() for path in file_paths}
+
+    cross_seeded_torrents: List[str] = []
+
+    # Define transient/network-related errors we treat conservatively
+    transient_errors: Tuple[type, ...] = (OSError, socket.timeout)
+    if qbittorrent_exceptions is not None:
+        # APIConnectionError covers various underlying network issues from qbittorrent-api
+        transient_errors = transient_errors + (qbittorrent_exceptions.APIConnectionError,)  # type: ignore
```

Ruff (0.14.6)

Consider iterable unpacking instead of concatenation

(RUF005)